### ISSUE TITLE
Error: 1228 - error during validation of incoming VPackObject index table is out of bounds

### ISSUE THREAD

**Issue by laihuei-n4** [arangodb/arangodb-java-driver#82]
Hi, I was using arangodb-java-driver 4.1.3 to create vertexes in ArangoDB with the method: com.arangodb.ArangoCollection.insertDocuments(Collection<BaseDocument> values).

In some cases, it worked well but sometimes it throws the exception below:

...
Caused by: com.arangodb.ArangoDBException: Response: 400, Error: 1228 - error during validation of incoming VPackObject index table is out of bounds
at com.arangodb.internal.velocystream.Communication.checkError(Communication.java:104)
at com.arangodb.internal.velocystream.CommunicationSync.execute(CommunicationSync.java:122)
at com.arangodb.internal.velocystream.CommunicationSync.execute(CommunicationSync.java:42)
at com.arangodb.internal.ArangoExecutorSync.execute(ArangoExecutorSync.java:58)
at com.arangodb.ArangoCollection.insertDocuments(ArangoCollection.java:115) 
...

The class I used for objects to store contains one field of long type. If I change the type to String, it works fine. Any idea?

**Comment by mpv1989** [comment]
Hi, can you please post one of the failing objects here?

Thanks & Best
Mark

**Comment by laihuei-n4** [comment]
I am not sure which objects causing the failure. Each time, the method insertDocuments is invoked for one chunk of 100 objects. Each object in ArangoDB UI looks like below. If I change long1 type from long to String, it works without the exception.

	map1		{2}
	list1		[4]
        string1	:	string-foo
        long1	:	71733115603202

**Comment by mpv1989** [comment]
Ok, thanks. I think the problem not only belongs to the long. As I see your objects contains additional fields. Can you post the Java code of your POJO and maybe a snippet of the initialization with some data?

Thanks

**Comment by laihuei-n4** [comment]
The exceptions are observed again even without data of long type. I am doing some testing to figure it out.

**Comment by miguel-porto** [comment]
I've got the very same issue with updateDocument:
https://github.com/arangodb/arangodb/issues/2226

**Comment by mpv1989** [comment]
It could be that this bug is already fixed in the actual 4.1.4-SNAPSHOT. Maybe you both can test it with your cases. Please let me know if this works. If it fixes the issue I will release a new version of the driver as soon as possible.

To get the snapshot with maven add the staging repository of oss.sonatype.org to your pom.xml:
```XML
<repositories>
  <repository>
    <id>arangodb-snapshots</id>
    <url>https://oss.sonatype.org/content/groups/staging</url>
  </repository>
</repositories>
```

Thanks & Best
Mark

**Comment by laihuei-n4** [comment]
After replacing 4.1.3 with 4.1.4-SNAPSHOT version, the issue doesn't exist in my test case. I have no idea what the root cause is and if this issue is fixed completely. Will continue observing. Thanks!

**Comment by miguel-porto** [comment]
I confirm: fixed with this version! ;-)

**Comment by mpv1989** [comment]
Ok, Thanks. I released version 4.1.4 with the fix.
